app/src/sharedTest/java/org/oppia/android/app/settings/profile/ProfileEditFragmentTest.kt

```diff
@@ -0,0 +1,591 @@
+    }
+  }
+
+  @Test
+  fun testProfileEdit_startWithUserHasDownloadAccess_clickAllowDownloadContainer_checkChanged() {
+    profileManagementController.addProfile(
+      name = "James",
+      pin = "123",
+      avatarImagePath = null,
+      allowDownloadAccess = true,
+      colorRgb = -10710042,
+      isAdmin = false
+    ).toLiveData()
+    launch<ProfileEditActivity>(
+      ProfileEditActivity.createProfileEditActivity(
+        context = context,
+        profileId = 4
+      )
+    ).use {
+      testCoroutineDispatchers.runCurrent()
+      onView(withId(R.id.profile_edit_allow_download_switch)).check(matches(isChecked()))
+      onView(withId(R.id.profile_edit_allow_download_container)).perform(click())
+      onView(withId(R.id.profile_edit_allow_download_switch)).check(matches(not(isChecked())))
+    }
+  }
+
+  @Test
+  fun testProfileEdit_startWithUserHasDownloadAccess_switchIsNotClickable() {
+    profileManagementController.addProfile(
+      name = "James",
+      pin = "123",
+      avatarImagePath = null,
+      allowDownloadAccess = true,
+      colorRgb = -10710042,
+      isAdmin = false
+    ).toLiveData()
+    launch<ProfileEditActivity>(
+      ProfileEditActivity.createProfileEditActivity(
+        context = context,
+        profileId = 4
+      )
+    ).use {
+      testCoroutineDispatchers.runCurrent()
+      onView(withId(R.id.profile_edit_allow_download_switch)).check(matches(not(isClickable())))
+    }
+  }
+
+  @Test
+  fun testProfileEdit_startWithUserHasDownloadAccess_switchContainerIsFocusable() {
+    profileManagementController.addProfile(
+      name = "James",
+      pin = "123",
+      avatarImagePath = null,
+      allowDownloadAccess = true,
+      colorRgb = -10710042,
+      isAdmin = false
+    ).toLiveData()
+    launch<ProfileEditActivity>(
+      ProfileEditActivity.createProfileEditActivity(
+        context = context,
+        profileId = 4
+      )
+    ).use {
+      testCoroutineDispatchers.runCurrent()
+      onView(withId(R.id.profile_edit_allow_download_container)).check(matches(isFocusable()))
+    }
+  }
+
+  @Test
+  fun testProfileEdit_startWithUserHasDownloadAccess_switchContainerIsDisplayed() {
+    profileManagementController.addProfile(
+      name = "James",
+      pin = "123",
+      avatarImagePath = null,
+      allowDownloadAccess = true,
+      colorRgb = -10710042,
+      isAdmin = false
+    ).toLiveData()
+    launch<ProfileEditActivity>(
+      ProfileEditActivity.createProfileEditActivity(
+        context = context,
+        profileId = 4
+      )
+    ).use {
+      testCoroutineDispatchers.runCurrent()
+      onView(withId(R.id.profile_edit_allow_download_container)).check(matches(isDisplayed()))
+    }
+  }
+
+  @Test
+  fun testProfileEdit_startWithUserDoesNotHaveDownloadAccess_switchContainerIsNotDisplayed() {
+    launch<ProfileEditActivity>(
+      ProfileEditActivity.createProfileEditActivity(
```

Comment:
If this suite aims to test ProfileEditFragment, it ought to test the fragment in isolation by wrapping it in a test activity (similar to how we do this for other fragments).